source/slang/slang-ir-peephole.cpp

```diff
@@ -250,6 +251,27 @@ struct PeepholeContext : InstPassBase
 
         switch (inst->getOp())
         {
+        case kIROp_AlignOf:
+            // Fold all calls to builtin functions that results in simple integer values.
+            // When postInlining flag is set, this happens for alignOf<T>() function, which
+            // infers the size of T and replaces the use with the IntLiteral.
+            if (isPostInlining && inst->getDataType()->getOp() == kIROp_IntType)
```

Comment:
Why do we need to protect this with `isPostInlining`? I see no problems with just running them regardless of pre or post linking. If any of the operands isn't of the right type or isn't a constant, we just skip any changes.